packages/protobuf21/PKGBUILD

```diff
@@ -0,0 +1,66 @@
+# Maintainer: OSAMC <https://github.com/osam-cologne/archlinux-proaudio>
+# Contributor: Lukas Fleischer <lfleischer@archlinux.org>
+# Contributor: David Runge <dvzrv@archlinux.org>
+# Contributor: Aleksey Filippov <sarum9in@gmail.com>
+# Contributor: Bartłomiej Piotrowski <bpiotrowski@archlinux.org>
+# Contributor: Sven-Hendrik Haase <sh@lutzhaase.com>
+# Contributor: Thomas S Hatch <thatch45@gmail.com>
+# Contributor: Geoffroy Carrier <geoffroy@archlinux.org>
+# Contributor: Daniel J Griffiths <ghost1227@archlinux.us>
+# Contributor: Tobias Boesch <tobias.boesch at googlemail.com>
+# Contributor: Florian Hülsmann <fh@cbix.de>
+
+_name=protobuf
+pkgname=protobuf21
+# DO NOT UPDATE, the point of this package is to pin the version
+pkgver=21.12
+pkgrel=1
+pkgdesc='Protocol Buffers (legacy)'
+arch=(aarch64 x86_64)
+url='https://developers.google.com/protocol-buffers/'
+license=(BSD)
+depends=(gcc-libs glibc zlib)
+makedepends=(cmake python-build python-installer python-setuptools python-wheel)
+source=("https://github.com/protocolbuffers/$_name/archive/v$pkgver/$_name-$pkgver.tar.gz"
+        "python3.11.patch::https://github.com/protocolbuffers/$_name/commit/da973aff.patch"
+        "fix-abi.patch::https://github.com/protocolbuffers/$_name/commit/101b6199.patch")
+sha256sums=('22fdaf641b31655d4b2297f9981fa5203b2866f8332d3c6333f6b0107bb320de'
+            '15ac9215cbbe1648dd92a0c4fc9ac3088d815bc5d86b8b1989cdc402a5f78ec5'
+            '96c4f97f662589d5daa0857ad47f740853cf273f817de21612383d76e4243b44')
+
+prepare() {
+  cd $_name-$pkgver
+  # Revert ABI break
+  patch -Rp1 < ../fix-abi.patch
+  # fix for python 3.11
+  patch -Np1 -i ../python3.11.patch
+}
+
+build() {
+  local cmake_options=(
+    -B build
+    -D CMAKE_BUILD_TYPE=None
+    -D CMAKE_INSTALL_PREFIX=/usr
+    -D CMAKE_INSTALL_INCLUDEDIR=/usr/include/$pkgname
+    -D protobuf_BUILD_SHARED_LIBS=ON
+    -D protobuf_BUILD_TESTS=OFF
+    -S $_name-$pkgver
+    -W no-dev
+  )
+
+  cmake "${cmake_options[@]}"
+  cmake --build build --verbose
+
+  ln -rs build $_name-$pkgver/src/.libs
+  cd $_name-$pkgver/python
+  PROTOC="$srcdir"/build/protoc \
+    python -m build --wheel --no-isolation
+}
+
+package_protobuf21() {
+  provides=("protobuf=$pkgver" libprotoc.so libprotobuf.so libprotobuf-lite.so)
+  DESTDIR="$pkgdir" cmake --install build
+  cd "$pkgdir"
+  mv usr/bin/protoc{,21}
+  rm -r usr/lib/{cmake,pkgconfig,*.so}
```

Comment:
How does software wanting to compile against this version of the lib find it without the pkgconf file?

Will it rely on the pkgconf files from the package for the current version?

Reply:
I previously just moved that to `usr/lib/protobuf21/pkgconfig`, but wasn't sure how to test if it works. For ola (the reason we need this package in the first place) they just check for some header files, anyway building doesn't work unless we also package an earlier `abseil-cpp` :/

Guess I'll go with renaming it again :)

Reply:
I'm not sure renaming would be sufficient, you probably also need to change the line

    Libs: -L${libdir} -lprotobuf

in `protobuf.pc` to pass the correct version for the `-l` option. Maybe have  look how other compatibility packages do it?

But if this package is only used for compiling `ola`, I'm not sure it's worth the effort. I just want to make sure that this package does not depend on the normal `protobuf`  package.

Reply:
One example I found was [podofo-0.9](https://gitlab.archlinux.org/archlinux/packaging/packages/podofo-0.9/-/blob/main/PKGBUILD), which is [used by scribus](https://gitlab.archlinux.org/archlinux/packaging/packages/scribus/-/blob/main/PKGBUILD) by passing the include/lib paths during build, even though there is a pkgconfig under `/usr/lib/podofo-0.9/pkgconfig`.

Well, also most "legacy" packages I found provide a custom libdir and link the versioned `.so` to `/usr/lib`. I'll do that again here.

Reply:
87c8feca3223f00ca4cdfad978f6904d8711c642